src/hibernate-resume/hibernate-resume-config.c

```diff
@@ -110,30 +108,20 @@ static int get_kernel_hibernate_location(KernelHibernateLocation **ret) {
 
 #if ENABLE_EFI
 static bool validate_efi_hibernate_location(EFIHibernateLocation *e) {
-        _cleanup_free_ char *id = NULL, *image_id = NULL, *version_id = NULL, *image_version = NULL;
-        struct utsname uts = {};
+        _cleanup_free_ char *id = NULL, *image_id = NULL;
         int r;
 
         assert(e);
 
-        if (uname(&uts) < 0)
-                log_warning_errno(errno, "Failed to get kernel info, ignoring: %m");
-
         r = parse_os_release(NULL,
                              "ID", &id,
-                             "IMAGE_ID", &image_id,
-                             "VERSION_ID", &version_id,
-                             "IMAGE_VERSION", &image_version);
+                             "IMAGE_ID", &image_id);
         if (r < 0)
-                log_warning_errno(r, "Failed to parse os-release, ignoring: %m");
+                log_warning_errno(r, "Failed to parse os-release: %m");
 
-        if (!streq(uts.release, strempty(e->kernel_version)) ||
-            !streq_ptr(id, e->id) ||
-            !streq_ptr(image_id, e->image_id) ||
-            !streq_ptr(version_id, e->version_id) ||
-            !streq_ptr(image_version, e->image_version)) {
-
-                log_notice("HibernateLocation system info doesn't match with current running system, not resuming from it.");
+        if (!streq_ptr(id, e->id) ||
+            !streq_ptr(image_id, e->image_id)) {
+                log_notice("HibernateLocation system identifier doesn't match currently running system, not resuming from it.");
                 return false;
         }
 
```

Comment:
i think it would make sense if we add a comment here that clarifies why we don't care for kernel version mismatches, because I am sure a future reader (for example: me) might get confused by that. i.e. something like:

```
/* Note that we accept kernel version mismatches, as Linux writes the old kernel to disk as part of the hibernation image, and thus resuming means the short-lived kernel that reads the image off disk will be replaced by the original kernel, and is removed from memory as part of that. */